equinox/nn/_composed.py

```diff
@@ -136,32 +140,62 @@ class Sequential(Module):
         Activation functions can be added by wrapping them in [`equinox.nn.Lambda`][].
     """
 
-    layers: tuple[_SequentialLayer, ...]
+    layers: tuple[Union[_SequentialLayer, _StatefulLayer], ...]
```

Comment:
I think I'd suggest just removing the `_SequentialLayer` protocol. IMO with this update it becomes more confusing than it is useful.

Reply:
Wouldn't that imply that layers now only accepts `StatefulLayer` ?

Reply:
I mean to remove both from type annotations: just have `layers: tuple`.